source/common/event/guarddog_impl.cc

```diff
@@ -0,0 +1,100 @@
+#include "common/common/assert.h"
+#include "common/event/guarddog_impl.h"
+#include "common/event/watchdog_impl.h"
+
+namespace Event {
+
+GuardDogImpl::GuardDogImpl(Stats::Store& stats_store, const Server::Configuration::Main& config,
+                           SystemTimeSource& tsource)
+    : time_source_(tsource), miss_timeout_(config.wdMissTimeout()),
+      megamiss_timeout_(config.wdMegaMissTimeout()), kill_timeout_(config.wdKillTimeout()),
+      multi_kill_timeout_(config.wdMultiKillTimeout()),
+      loop_interval_(
+          std::min({miss_timeout_, megamiss_timeout_,
+                    kill_enabled() ? kill_timeout_ : std::min(miss_timeout_, megamiss_timeout_),
```

Comment:
The repetition of `std::min(miss_timeout_, ..)` could be avoided if this was done inside the constructor rather than initializor list. You can simplify it a bit, since it's min and you just need something here >= miss_timeout_/megamiss_timeout_.